cmake/Dependencies.cmake

```diff
@@ -0,0 +1,99 @@
+set(LIBRARY_LINKAGE SHARED)
+
+if ("${BOOST_ROOT_DIR}" STREQUAL "")
+  set (BOOST_INCL "${MESOS_ROOT}/build/3rdparty/boost-${BOOST_VERSION}/src/boost-${BOOST_VERSION}")
+else ()
+  set (BOOST_INCL ${BOOST_ROOT_DIR}/include)
+endif ()
+
+set (PICOJSON_INCL
+  "${MESOS_ROOT}/include"
+  "${MESOS_ROOT}/build/3rdparty/picojson-${PICOJSON_VERSION}/src/picojson-${PICOJSON_VERSION}")
+set (RAPIDJSON_INCL
+  "${MESOS_ROOT}/include"
+  "${MESOS_ROOT}/build/3rdparty/rapidjson-${RAPIDJSON_VERSION}/src/rapidjson-${RAPIDJSON_VERSION}/include")
+set (STOUT_INCL
+  "${MESOS_ROOT}/include"
+  "${MESOS_ROOT}/3rdparty/stout/include")
+set (ZOOKEEPER_INCL
+  "${MESOS_ROOT}/include/zookeeper"
+  "${MESOS_ROOT}/build/3rdparty/zookeeper-${ZOOKEEPER_VERSION}/src/zookeeper-${ZOOKEEPER_VERSION}/src/c/generated"
+  "${MESOS_ROOT}/build/3rdparty/zookeeper-${ZOOKEEPER_VERSION}/src/zookeeper-${ZOOKEEPER_VERSION}/src/c/include")
+
+set (GLOG_INCL
+  "${MESOS_ROOT}/include"
+  "${MESOS_ROOT}/build/3rdparty/glog-${GLOG_VERSION}/src/glog-${GLOG_VERSION}-install/include")
+find_library(
```

Comment:
If we end up relying on `-DENABLE_INSTALL_MODULE_DEPENDENCIES` (I think we should), then we should add `NO_DEFAULT_PATH` in all invocations of find_library()  for bundled libraries.

Rationale: developing mesos modules is much simpler when you don't have to think: "Damn, I have glog installed on my system, which glog were the modules built against? was it the bundled one or the system one?..."

Reply:
right... awaiting votes/decision about this topic in other PR.

Reply:
We settled on using -DENABLE_INSTALL_MODULE_DEPENDENCIES, this means that we can add `NO_DEFAULT_PATH`, right?

Reply:
yes, added `NO_DEFAULT_PATH` to each of `find_library` in `Deendencies.cmake`.